File: lib/JFFUtils/NSString/NSString+Trimm.m
#import "NSString+Trimm.h"

@implementation NSString (Trimm)

-(NSRange)rangeForQuotesRemoval
{
    NSString *quotedString = self;
    
    static const NSUInteger firstQuoteOffset = 1;
    static const NSUInteger quotesCount = 2;
    NSUInteger rangeLength = [quotedString length] - quotesCount;
    
    NSRange result = {firstQuoteOffset, rangeLength};
    return result;
}

- (NSString *)stringByTrimmingWhitespaces
{
    NSCharacterSet *set = [NSCharacterSet whitespaceAndNewlineCharacterSet];
    return [self stringByTrimmingCharactersInSet:set];
}

- (NSString *)stringByTrimmingPunctuation
{
    NSCharacterSet *set = [NSCharacterSet punctuationCharacterSet];
    return [self stringByTrimmingCharactersInSet:set];
}

- (NSString *)stringByTrimmingQuotes
{
    NSRange rangeWithoutQuotes = [self rangeForQuotesRemoval];
    NSString *result = [self substringWithRange:rangeWithoutQuotes];
    
    NSCharacterSet *termWhitespaces = [NSCharacterSet whitespaceAndNewlineCharacterSet];
    
    return [result stringByTrimmingCharactersInSet:termWhitespaces];
}

@end

File: iAsync.Swift/lib/JUtils/JUtils/String/NSString+Trimm.swift
//
//  NSString+Trimm.swift
//  JUtils
//
//  Created by Vladimir Gorbenko on 08.06.14.
//  Copyright (c) 2014 EmbeddedSources. All rights reserved.
//

import Foundation

public extension NSString {
    
    private func rangeForQuotesRemoval() -> NSRange {
        
        let quotedString = self
        
        let firstQuoteOffset = 1
        let quotesCount      = 2
        let rangeLength = quotedString.length - quotesCount
        
        let result = NSMakeRange(firstQuoteOffset, rangeLength)
        return result
    }
    
    func stringByTrimmingWhitespaces() -> String {
        
        let set = NSCharacterSet.whitespaceAndNewlineCharacterSet()
        return stringByTrimmingCharactersInSet(set)
    }
    
    func stringByTrimmingPunctuation() -> String  {
        
        let set = NSCharacterSet.punctuationCharacterSet()
        return stringByTrimmingCharactersInSet(set)
    }
    
    func stringByTrimmingQuotes() -> NSString {
        
        let rangeWithoutQuotes = rangeForQuotesRemoval()
        let result = substringWithRange(rangeWithoutQuotes)
        
        let termWhitespaces = NSCharacterSet.whitespaceAndNewlineCharacterSet()
        
        return result.stringByTrimmingCharactersInSet(termWhitespaces)
    }
}
